user-operator/src/main/java/io/strimzi/operator/user/Main.java

```diff
@@ -99,7 +100,7 @@ public static void main(String[] args) {
         );
 
         // Create the health check and metrics server
-        HealthCheckAndMetricsServer healthCheckAndMetricsServer = new HealthCheckAndMetricsServer(controller, metricsProvider);
+        HealthCheckAndMetricsServer healthCheckAndMetricsServer = new HealthCheckAndMetricsServer(controller, controller, metricsProvider);
```

Comment:
We could create a common interface that extends Liveness, Readiness and MetricsProvider. Then we could use it as an additional one-argument constructor for the HealthCheckAndMetricsServer. WDYT?